src/Microsoft.Health.Fhir.Api/Features/SMART/SmartClinicalScopesMiddleware.cs

```diff
@@ -77,84 +77,87 @@ public SmartClinicalScopesMiddleware(RequestDelegate next, ILogger<SmartClinical
 
                     // examine the scopes claim for any SMART on FHIR clinical scopes
                     DataActions permittedDataActions = 0;
-                    foreach (Claim claim in principal.FindAll(authorizationConfiguration.ScopesClaim))
+                    foreach (string singleScope in authorizationConfiguration.ScopesClaim)
```

Comment:
I think this foreach includes too much of the file.  It also wraps up the fhirUserClaim processing which doesn't need to be repeated multiple times.  It doesn't look like there is any harm in running it more than needed, but I feel this could be a source of errors later.  I think we should have this foreach iterate over all the claims in the list, and then perhaps build one concatenated string of the values from the different claims, then this longer string could be passed to the regex to identify all the scopes.